src/rdkafka_queue.c

```diff
@@ -563,13 +563,15 @@ rd_kafka_purge_outdated_messages(rd_kafka_toppar_t *rktp,
         }
 
         /* Discard outdated control msgs ops */
-        next = TAILQ_FIRST(&ctrl_msg_q);
-        while (next && next->rko_rktp == rktp &&
-               rd_kafka_op_version_outdated(next, version)) {
+        next = TAILQ_FIRST(ctrl_msg_q);
+        while (next) {
```

Comment:
I feel that we don't need to iterate full queue as all the higher versions are going to be at the end. But its fine this way as well just that we will process more ops.